File: sdk/python/dialect/recon_database_dialect.py
from abc import ABC, abstractmethod
import mysql.connector
import psycopg2


class ReconDatabaseDialect(ABC):
    """数据库方言抽象基类"""

    @abstractmethod
    def get_database_type(self) -> str:
        """获取数据库类型名称"""
        pass

    @abstractmethod
    def get_date_function(self) -> str:
        """获取日期函数"""
        pass

    @abstractmethod
    def get_current_time_function(self) -> str:
        """获取当前时间函数"""
        pass


class MySQLDialect(ReconDatabaseDialect):
    """MySQL方言实现"""

    def get_database_type(self) -> str:
        return "mysql"

    def get_date_function(self) -> str:
        return "DATE"

    def get_current_time_function(self) -> str:
        return "NOW()"


class PostgreSQLDialect(ReconDatabaseDialect):
    """PostgreSQL方言实现"""

    def get_database_type(self) -> str:
        return "postgresql"

    def get_date_function(self) -> str:
        return "DATE"

    def get_current_time_function(self) -> str:
        return "CURRENT_TIMESTAMP"


def create_dialect(connection) -> ReconDatabaseDialect:
    """创建数据库方言"""
    # 通过连接对象判断数据库类型
    if isinstance(connection, mysql.connector.connection.MySQLConnection):
        return MySQLDialect()
    elif isinstance(connection, psycopg2.extensions.connection):
        return PostgreSQLDialect()
    else:
        # 默认返回MySQL方言
        return MySQLDialect()

File: sdk/node.js/dialect/ReconDatabaseDialect.js
/**
 * 数据库方言抽象类
 */
class ReconDatabaseDialect {
  /**
   * 获取数据库类型名称
   */
  getDatabaseType() {
    throw new Error('子类必须实现getDatabaseType方法');
  }

  /**
   * 获取建表 SQL 语句
   * @returns {Array<string>}
   */
  getCreateTableSQL() {
    throw new Error('子类必须实现getCreateTableSQL方法');
  }

  /**
   * 获取日期函数
   */
  getDateFunction() {
    throw new Error('子类必须实现getDateFunction方法');
  }

  /**
   * 获取当前时间函数
   */
  getCurrentTimeFunction() {
    throw new Error('子类必须实现getCurrentTimeFunction方法');
  }
}

/**
 * 创建数据库方言
 * @param {Object} connection - 数据库连接对象
 * @returns {ReconDatabaseDialect} 数据库方言实例
 */
function createDialect(connection) {
  // 简单判断: mysql2 连接通常有 config 属性，pg 连接有 connectionParameters
  if (connection.config && connection.config.namedPlaceholders !== undefined) {
    const MySQLDialect = require('./MySQLDialect');
    return new MySQLDialect();
  } else if (connection.connectionParameters || (connection.options && connection.options.dialect === 'postgres')) {
    const PostgreSQLDialect = require('./PostgreSQLDialect');
    return new PostgreSQLDialect();
  } else {
    // 默认尝试 MySQL，或者抛出错误
    const MySQLDialect = require('./MySQLDialect');
    return new MySQLDialect();
  }
}

module.exports = {
  ReconDatabaseDialect,
  createDialect
};
